ui/hooks/useAccountGroupsForPermissions.ts

```diff
@@ -0,0 +1,131 @@
+import { useSelector } from 'react-redux';
+import {
+  Caip25CaveatValue,
+  getCaipAccountIdsFromCaip25CaveatValue,
+} from '@metamask/chain-agnostic-permission';
+import { CaipChainId, CaipNamespace } from '@metamask/utils';
+import { useMemo } from 'react';
+import {
+  getCaip25AccountIdToMultichainAccountGroupMap,
+  getAccountGroupWithInternalAccounts,
+  getScopeToAccountGroupMap,
+} from '../selectors/multichain-accounts/account-tree';
+import { AccountGroupWithInternalAccounts } from '../selectors/multichain-accounts/account-tree.types';
+
+/**
+ * Hook that manages account groups for CAIP-25 permissions, providing both connected
+ * and supported account groups based on existing permissions and requested chains/namespaces.
+ *
+ * This hook handles the complex logic of:
+ * - Filtering account groups that support requested chains/namespaces
+ * - Mapping existing CAIP-25 permissions to account groups
+ * - Converting between different account ID formats
+ *
+ * @param existingPermission - The current CAIP-25 caveat value containing existing permissions
+ * @param requestedCaipChainIds - Array of CAIP chain IDs being requested for permission
+ * @param requestedNamespacesWithoutWallet - Array of CAIP namespaces being requested (excluding wallet namespace)
+ * @returns Object containing connected account groups, supported account groups, and existing connected CAIP account IDs
+ */
+export const useAccountGroupsForPermissions = (
```

Comment:
@montelaidev lets make sure to test how this hook+selectors perform with power users (+50 multichain accounts)